### PR TITLE
Allow RocketChatAttachment class accept DateTimeImmutable as timestamp argument

### PR DESCRIPTION
**Pull request by antonkomarev**
This PR adds failing tests for DateTimeImmutable timestamp. This need to be fixed.

### CODE REVIEW

**Comment by antonkomarev** [comment]
I think we could separate this method on 2 strict typed methods:
- `timestamp(DateTimeInterface $dateTime)`
- `timestampFromString(string $dateTime)`

**Comment by atymic** [comment]
It's pretty common for laravel to accept mixed types (eg, a date string or a `DateTimeInterface`).
I generally end up checking if it's a `DateTimeInterface` and falling back to parsing as a string.

**Comment by antonkomarev** [comment]
@atymic yeah, but `DateTimeInterface` doesn't have `setTimezone` method:
https://www.php.net/manual/en/class.datetimeinterface.php

**Comment by Funfare** [comment]
RocketChat also accepts DATE_ATOM timestamp, so I changed the code to using that. Then DateTimeInterface works fine

File: src/RocketChatAttachment.php [hunk 1]
```diff
@@ -103,18 +102,25 @@ public function text(string $text): self
...
+
+            throw new InvalidArgumentException(sprintf(
+                'Timestamp must be string or DateTime, %s given.',
+                $invalidType
+            ));
         }
 
-        if ($timestamp instanceof DateTime) {
+        if ($timestamp instanceof DateTimeInterface) {
             $date = clone $timestamp;
```
Comment (antonkomarev) [inline src/RocketChatAttachment.php]: I think we don't need `clone` object here, because we are not mutating it anymore. We could just format it.

**Comment by antonkomarev** [comment]
I've replaced ATOM with RFC3339 because its name of the standard (their values are same).

**Pull request merged**